packages/kotti-ui/source/kotti-filter/utils.ts

```diff
@@ -0,0 +1,114 @@
+import { Kotti } from '../types'
+
+export const getEmptyFilter = (
+	columnKey: Kotti.Filter.InternalColumn['key'],
+	columns: Kotti.Filter.InternalColumn[],
+): Kotti.Filter.InternalFilter | null => {
+	const column = columns.find(
+		(columnItem: Kotti.Filter.InternalColumn) => columnItem.key === columnKey,
+	)
+	switch (column?.filterDefinition.type) {
+		case Kotti.Filter.FilterType.STRING:
+			return {
+				key: columnKey,
+				operation: Kotti.Filter.Operation.String.CONTAINS,
+				value: null,
+			}
+		case Kotti.Filter.FilterType.NUMBER:
+			return {
+				key: columnKey,
+				operation: Kotti.Filter.Operation.Number.EQUAL,
+				value: null,
+			}
+		case Kotti.Filter.FilterType.BOOLEAN:
+			return {
+				key: columnKey,
+				operation: Kotti.Filter.Operation.Boolean.EQUAL,
+				value: null,
+			}
+		case Kotti.Filter.FilterType.DATE_RANGE:
+			return {
+				key: columnKey,
+				operation: Kotti.Filter.Operation.DateRange.IN,
+				value: [null, null],
+			}
+		case Kotti.Filter.FilterType.SINGLE_ENUM:
+			return {
+				key: columnKey,
+				operation: Kotti.Filter.Operation.SingleEnum.EQUAL,
+				value: null,
+			}
+		case Kotti.Filter.FilterType.MULTI_ENUM:
+			return {
+				key: columnKey,
+				operation: Kotti.Filter.Operation.MultiEnum.ONE_OF,
+				value: [],
+			}
+		default:
+			return null
+	}
+}
+
+export const getOperationOptions = (
+	filterType: Kotti.Filter.FilterType,
+): Kotti.FieldSingleSelect.Props['options'] => {
+	switch (filterType) {
+		case Kotti.Filter.FilterType.STRING:
+			return Object.values(Kotti.Filter.Operation.String).map((operation) => ({
+				label: Kotti.Filter.OperationLabel.String[operation],
+				value: operation,
+			}))
+		case Kotti.Filter.FilterType.NUMBER:
+			return Object.values(Kotti.Filter.Operation.Number).map((operation) => ({
+				label: Kotti.Filter.OperationLabel.Number[operation],
+				value: operation,
+			}))
+		case Kotti.Filter.FilterType.BOOLEAN:
+			return Object.values(Kotti.Filter.Operation.Boolean).map((operation) => ({
+				label: Kotti.Filter.OperationLabel.Boolean[operation],
+				value: operation,
+			}))
+		case Kotti.Filter.FilterType.DATE_RANGE:
+			return Object.values(Kotti.Filter.Operation.DateRange).map(
+				(operation) => ({
+					label: Kotti.Filter.OperationLabel.DateRange[operation],
+					value: operation,
+				}),
+			)
+		case Kotti.Filter.FilterType.SINGLE_ENUM:
+			return Object.values(Kotti.Filter.Operation.SingleEnum).map(
+				(operation) => ({
+					label: Kotti.Filter.OperationLabel.SingleEnum[operation],
+					value: operation,
+				}),
+			)
+		case Kotti.Filter.FilterType.MULTI_ENUM:
+			return Object.values(Kotti.Filter.Operation.MultiEnum).map(
+				(operation) => ({
+					label: Kotti.Filter.OperationLabel.MultiEnum[operation],
+					value: operation,
+				}),
+			)
+		default:
+			return []
```

Comment:
When does the `default` case ever intentionally happen? This should be omitted to allow TypeScript to find the bug if someone adds a new filter type and forgets to add the empty state here. Alternatively, it should throw a runtime Error via `throw new Error()`